lib/ansible/modules/cloud/memset/memset_lb_service.py

```diff
@@ -0,0 +1,342 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2019, Simon Weald <ansible@simonweald.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: memset_lb_service
+author: "Simon Weald (@glitchcrab)"
+version_added: "2.9"
+short_description: Manage Memset loadbalancer services
+notes:
+  - A loadbalancer service is logically the Internet-facing 'frontend'. This must
+    be backed by one or more servers using the C(memset_lb_server) module.
+  - An API key generated via the Memset customer control panel is needed with the
+    following minimum scope - I(loadbalancer.service.add), I(loadbalancer.service.info),
+    I(loadbalancer.service.list), I(loadbalancer.service.remove),
+    I(loadbalancer.service.update), I(server.info).
+description:
+    - Manage Memset loadbalancer services
+options:
+    state:
+        required: false
+        default: present
+        description:
+            - Indicates desired state of resource. Defaults to present.
+            - When deleting a service there must be no servers currently attached to it as this will raise an error.
+        choices: [ absent, present ]
+    api_key:
+        required: true
+        description:
+            - The API key obtained from the Memset control panel.
+    enabled:
+        required: false
+        default: true
+        type: bool
+        description:
+            - Whether the service is enabled or not. Defaults to True.
+    load_balancer:
+        required: true
+        description:
+            - The name of the load balancer - this is the product name e.g. C(lbtestyaa1).
+    port:
+        required: true
+        description:
+            - The port to be exposed to the Internet.
+            - Must be in the range 1 > 65535 (inclusive).
+    protocol:
+        required: true
```

Comment:
to me it looks that protocol should only be required if state=present --> see required_if helper